File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.cpp
#include <string>
#include <iostream>
#include <map>
using namespace std;

bool solution(string s)
{
    bool answer = true;
    
    map<char, int> hash;
    for(const auto &it : s){
        if(it == 'P'){
            hash['p'] += 1;
        }
        else if(it == 'Y'){
            hash['y'] += 1;
        }
        else{
            hash[it] += 1;
        }
    }
        
    return hash['p'] == hash['y'] ? true : false;
}

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.java
class Solution {
   boolean solution(String s) {
        boolean answer = true;  
        
        s = s.toLowerCase();

       int ps = s.length() - s.replace("p", "").length();
       int ys = s.length() - s.replace("y", "").length();

        return ps == ys;
    }

}